am-lib/src/main/java/uk/gov/hmcts/reform/amlib/DefaultRoleSetupImportService.java

```diff
@@ -0,0 +1,211 @@
+package uk.gov.hmcts.reform.amlib;
+
+import lombok.NonNull;
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.sqlobject.SqlObjectPlugin;
+import uk.gov.hmcts.reform.amlib.enums.AccessType;
+import uk.gov.hmcts.reform.amlib.enums.RoleType;
+import uk.gov.hmcts.reform.amlib.enums.SecurityClassification;
+import uk.gov.hmcts.reform.amlib.exceptions.ErrorAddingEntriesException;
+import uk.gov.hmcts.reform.amlib.exceptions.ErrorWithDeletionException;
+import uk.gov.hmcts.reform.amlib.models.DefaultPermission;
+import uk.gov.hmcts.reform.amlib.models.DefaultPermissionGrant;
+import uk.gov.hmcts.reform.amlib.models.ResourceAttribute;
+import uk.gov.hmcts.reform.amlib.repositories.DefaultRoleSetupRepository;
+import uk.gov.hmcts.reform.amlib.utils.Permissions;
+
+import javax.sql.DataSource;
+
+public class DefaultRoleSetupImportService {
+    private final Jdbi jdbi;
+
+    public DefaultRoleSetupImportService(String url, String user, String password) {
+        this.jdbi = Jdbi.create(url, user, password);
+        this.jdbi.installPlugin(new SqlObjectPlugin());
+    }
+
+    public DefaultRoleSetupImportService(DataSource dataSource) {
+        this.jdbi = Jdbi.create(dataSource);
+        this.jdbi.installPlugin(new SqlObjectPlugin());
+    }
+
+    /**
+     * A method to add a new service.
```

Comment:
Usually javadoc does not start this way. No point saying `A method to`. Have a look at examples from the popular libraries e.g. Guava or Apache Commons.

Example:
> Ensures the truth of an expression involving one or more parameters to the calling method.

Also describe behaviour like an update if exists or transaction boundaries. Let's give people proper documentation please.